### DISCUSSION TITLE
Themes - a mess?

### DISCUSSION THREAD

**Discussion by drei97** [AvaloniaUI/Avalonia#14639] [Q&A]
Hi,

since TreeDataGrid still is lacking a simple theme, I am wondering if it is possible to create one easily by my own. The themes of DataGrid and TreeDataGrid should not be too different.

But what I found is kinda mess. Some extensions are xaml, some axaml. Sometimes default setting are coming before dark settings and sometimes the other way round. Even the order of entries in the files is not the same. Every creator is doing its own thing? That not must be bad but makes things more difficult as it should be.

Is there any guide on how theme file should be structured or is there a kind of template?

Why are (brush) settings for DataGrid DynamicResources and for TreeDataGrid Static?

I know that it is work in progress and TreeDataGrid is 11.0.2 and DataGrid is 11.0.9. But before I tie my brain into knots I want to ask for some basics I am maybe not aware of.

**Comment by thevortexcloud** [comment]
> Some extensions are xaml, some axaml.

Extension does not matter that much. `axaml` was chosen to make Visual Studio happy since it's hard coded to treat `xaml` as WPF.

**Reply by drei97** [reply]
really hard coded to WPF? Oh dear!

**Comment by maxkatz6** [comment]
> since TreeDataGrid still is lacking a simple theme

https://github.com/AvaloniaUI/Avalonia.Controls.TreeDataGrid/issues/246#issuecomment-1874758099
